includes/class-wc-payments-order-service.php

```diff
@@ -643,6 +650,29 @@ public function attach_intent_info_to_order( $order, $intent_id, $intent_status,
 		$order->save();
 	}
 
+	/**
+	 * Given the charge, adds the application_fee_amount from the charge to the given order as metadata.
+	 *
+	 * @param WC_Order               $order The order to update.
+	 * @param WC_Payments_API_Charge $charge The charge to get the application_fee_amount from.
+	 */
+	public function attach_transaction_fee_to_order( $order, $charge ) {
+		try {
+			if ( $charge && $charge->get_application_fee_amount() ) {
+				$order->add_meta_data(
```

Comment:
I think we can use update_meta_data, which can ensure that this meta key is unique.

_Edit:_ I see the same pattern of using update_meta_data for other meta keys. 

```suggestion
				$order->update_meta_data(
```

Reply:
Done